src/Workspaces/Core/Portable/Execution/AssetStorages.cs

```diff
@@ -70,18 +74,17 @@ public RemotableData GetRemotableData(int scopeId, Checksum checksum, Cancellati
 
             // search snapshots we have
             var storage = _storages[scopeId];
-            var remotableData = storage.TryGetRemotableData(checksum, cancellationToken);
+            var remotableData = await storage.TryGetRemotableDataAsync(checksum, cancellationToken).ConfigureAwait(false);
             if (remotableData != null)
             {
                 return remotableData;
             }
 
             // search global assets
-            foreach (var kv in _globalAssets)
+            foreach (var asset in _globalAssets.Values)
```

Comment:
:bulb: For consistency with the change below

```suggestion
            foreach (var (_, asset) in _globalAssets)
```